lib/live_view_native_swift_ui/types/animation.ex

```diff
@@ -0,0 +1,20 @@
+defmodule LiveViewNativeSwiftUi.Types.Animation do
+  @derive Jason.Encoder
+  defstruct [:type, :properties, :modifiers]
+
+  use LiveViewNativePlatform.Modifier.Type
+  def type, do: :map
+
+  def cast(value) when is_atom(value), do: {:ok, %__MODULE__{ type: value, properties: %{}, modifiers: [] }}
+  def cast({value, [ {k, _} | _ ] = properties}) when is_atom(value) and is_atom(k) do
+    {:ok, %__MODULE__{ type: value, properties: Enum.into(properties, %{}), modifiers: [] }}
+  end
+  def cast({value, [ {k, _} | _ ] = properties, modifiers}) when is_atom(value) and is_atom(k) and is_list(modifiers) do
+    {:ok, %__MODULE__{ type: value, properties: Enum.into(properties, %{}), modifiers: Enum.map(modifiers, &cast_modifier/1) }}
+  end
```

Comment:
Small nitpick: I think the binding `value` should be renamed to `type` for these two patterns to be more semantically correct:

```suggestion
  def cast({type, [ {k, _} | _ ] = properties}) when is_atom(type) and is_atom(k) do
    {:ok, %__MODULE__{ type: type, properties: Enum.into(properties, %{}), modifiers: [] }}
  end
  def cast({type, [ {k, _} | _ ] = properties, modifiers}) when is_atom(type) and is_atom(k) and is_list(modifiers) do
    {:ok, %__MODULE__{ type: type, properties: Enum.into(properties, %{}), modifiers: Enum.map(modifiers, &cast_modifier/1) }}
  end
```